tokio/src/net/tcp/stream.rs

```diff
@@ -983,6 +983,11 @@ impl TcpStream {
     /// defined on the Tokio `TcpStream` type, as this will mess with the
     /// readiness flag and can cause the socket to behave incorrectly.
     ///
+    /// This method is not intended to be used with combined interests.
+    /// The closure should perform only one type of IO operation, what typically
+    /// needs only one type of interest. The closure is not guaranteed to be called
+    /// if the provided interest is combined.
```

Comment:
I think this phrasing works better.
```suggestion
    /// This method is not intended to be used with combined interests.
    /// The closure should perform only one type of IO operation, so it should
    /// not be called with a combined interest. The call may panic or sleep
    /// forever if it is called with a combined interest.
```